### ISSUE TITLE
Recent addition of checkstyle fails the build

### ISSUE THREAD

**Issue by ctron** [eclipse/kura#613]
The recent addition of checkstyle fails the build on hudson.eclipse.org. And it doesn't look to me as if this is on purpose.

```
[INFO] o.h.m.e.h.MavenExecutionResultHandler - Build failed with exception(s)
[INFO] o.h.m.e.h.MavenExecutionResultHandler - [1] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.14:check (checkstyle-validation) on project kura: Execution checkstyle-validation of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.14:check failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-checkstyle-plugin:2.14:check: java.lang.NoSuchMethodError: org.slf4j.spi.LocationAwareLogger.log(Lorg/slf4j/Marker;Ljava/lang/String;ILjava/lang/String;Ljava/lang/Throwable;)V
-----------------------------------------------------
realm =    plugin>org.apache.maven.plugins:maven-checkstyle-plugin:2.14
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
```

@MMaiero please have a look

**Comment by MMaiero** [comment]
It works fine in my environment.
I can find two possible solutions: stick on plugin version 2.14 and try to solve the issue like specified [here](http://stackoverflow.com/questions/15874503/maven-checkstyle-plugin-and-guava-google-collections-dependency-clash) or try to upgrade to 2.17 and see if the problem is still the same. I'll try with this last since the CQ already in place for 2.14 is not useful for this build-time dependency

**Comment by ctron** [comment]
AFAIK Kura currently doesn't have any build & test dependencies CQ. So it doesn't matter.

Unless I missed something, please try upgrading to 2.17 since it currently does fails the main build on the Hudson instance.

**Comment by MMaiero** [comment]
it seems not to be the plugin version the cause of the issue in hudson

**Comment by ctron** [comment]
Yes. So I suggest you put it in a profile and enable it only for the travis build at the moment.

**Comment by ctron** [comment]
Any maybe just directly push to `develop`

**Comment by MMaiero** [comment]
ok, now the build does not seem to fail due to the check style plugin

`[INFO] o.h.m.e.h.MavenExecutionResultHandler - Build failed with exception(s)
[INFO] o.h.m.e.h.MavenExecutionResultHandler - [1] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.20.0:compile (default-compile) on project org.eclipse.kura.core: Compilation failure
[DEBUG] Closing connection to remote
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.20.0:compile (default-compile) on project org.eclipse.kura.core: Compilation failure: Compilation failure:
[ERROR] /jobs/genie.kura/kura-develop/workspace/kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/pool/KuraJDBCPooledDatasource.java:[69]
[ERROR] public Logger getParentLogger() throws SQLFeatureNotSupportedException {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method getParentLogger() of type KuraJDBCPooledDatasource must override or implement a supertype method
[ERROR] 1 problem (1 error)
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.20.0:compile (default-compile) on project org.eclipse.kura.core: Compilation failure
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: copied.org.apache.maven.plugin.CompilationFailureException: Compilation failure
	at copied.org.apache.maven.plugin.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:441)
	at org.eclipse.tycho.compiler.AbstractOsgiCompilerMojo.execute(AbstractOsgiCompilerMojo.java:237)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more`

**Comment by ctron** [comment]
Yes, now it fails due to the BREE validation. Which is correct. I can now fix issue #592 

As posted on the mailing list today, the BREE will now be validated and will fail the build is validation fails.
